src/main/java/daos/DBUserDataAccessObject.java

```diff
@@ -1,39 +1,158 @@
 package daos;
 
+import entity.CommonUser;
 import entity.User;
+import entity.UserFactory;
```

Comment:
I think I might have the same issue in my PR, but shouldn't the DAO layer be decoupled from entities to follow CA rules?

Should we have an intermediate layer that converts our User entities to MongoDB schemas?

Reply:
I've replied to ur comment in that PR, I'll check with the TA in lab on monday, make a change and merge it in